spec/overcommit/hook/pre_commit/php_stan_spec.rb

```diff
@@ -0,0 +1,50 @@
+require 'spec_helper'
+
+describe Overcommit::Hook::PreCommit::PhpStan do
+  let(:config)  { Overcommit::ConfigurationLoader.default_configuration }
+  let(:context) { double('context') }
+  subject { described_class.new(config, context) }
+
+  before do
+    subject.stub(:applicable_files).and_return(%w[sample.php])
+  end
+
+  context 'when phpstan exits successfully' do
+    before do
+      sample_output = [
+        ''
+      ].join("\n")
+
+      result = double('result')
+      result.stub(:success?).and_return(true)
+      result.stub(:stdout).and_return(sample_output)
+      result.stub(:status).and_return(0)
+      subject.stub(:execute).and_return(result)
+    end
+
+    it { should pass }
+  end
+
+  context 'when phpstan exits unsuccessfully' do
+    let(:result) { double('result') }
+
+    before do
+      result.stub(:success?).and_return(false)
+      result.stub(:status).and_return(2)
+      subject.stub(:execute).and_return(result)
+    end
+
+    context 'and it reports a warning' do
+      before do
+        sample_output = [
+          '/sample1.php:14:Call to an undefined static method Sample1::where()',
+          '/sample2.php:17:Anonymous function has an unused use $myVariable.'
+        ].join("\n")
+        # rubocop:enable Metrics/LineLength
```

Comment:
Why is this comment here? It doesn't seem to be necessary.

Reply:
Opps. Will remove it.